Documentation/ApiOverview/Http/Index.rst

```diff
@@ -7,90 +7,63 @@
 
 .. _http:
 
-HTTP request library
---------------------
+HTTP request library / Guzzle / PSR-7
+-------------------------------------
 
-Since TYPO3 CMS 4.6, a library for easily making HTTP requests
-is available. It is actually a wrapper around the
-`HTTP_Request2 PEAR package <http://pear.php.net/manual/en/package.http.http-request2.php>`_,
-which is shipped with the Core.
+Since TYPO3 CMS 8.1 the PHP library `Guzzle` has been added via composer dependency
+to work as a feature rich solution for creating HTTP requests based on the PSR-7 interfaces
+already used within TYPO3.
 
+Guzzle auto-detects available underlying adapters available on the system, like cURL or
+stream wrappers and chooses the best solution for the system.
 
-.. _http-basic:
-
-Basic usage
-^^^^^^^^^^^
-
-The basic usage is as simple as it gets:
+A TYPO3-specific PHP class called `TYPO3\CMS\Core\Http\RequestFactory` has been added as
+a simplified wrapper to access Guzzle clients.
 
-.. code-block:: php
+All options available under `$TYPO3_CONF_VARS[HTTP]` are automatically applied to the Guzzle
+clients when using the `RequestFactory` class. The options are a subset to the available options
+on Guzzle (http://docs.guzzlephp.org/en/latest/request-options.html) but can further be extended.
 
-   $request = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(
-      'TYPO3\\CMS\\Core\\Http\\HttpRequest',
-      'http://typo3.org/'
-   );
-   $result = $request->send();
-   $content = $result->getBody();
+Existing `$TYPO3_CONF_VARS[HTTP]` options have been removed and/or migrated to the
+new Guzzle-compliant options.
 
+A full documentation for Guzzle can be found at http://docs.guzzlephp.org/en/latest/.
 
-The above example will read the content of the "typo3.org" home page.
+Although Guzzle can handle Promises/A+ and asynchronous requests, it currently acts as
+a drop-in replacement for the previous mixed options and implementations within
+`GeneralUtility::getUrl()` and a PSR-7-based API for HTTP requests.
 
+The existing TYPO3-specific wrapper `GeneralUtility::getUrl()` now uses Guzzle under the hood
+automatically for remote files, removing the need to configure settings based on certain
+implementations like stream wrappers or cURL directly.
 
-.. _http-basic-example:
+.. _http-basic:
```

Comment:
Just good to know: This change seems ok. Though I would make it '.. _http-Basic-usage:'. Rule: Repeat the headline, replace blanks by a minus. Case doesn't matter. Note: You wouldn't change a label once it's being used by people and really important. But you can use more than one. Just add another line.